clang/test/CodeGenSYCL/intel-max-global-work-dim.cpp

```diff
@@ -26,12 +29,17 @@ int main() {
 
     Functor<2> f;
     h.single_task<class kernel_name3>(f);
+
+    h.single_task<class kernel_name4>([]() {
+      func<2>();
```

Comment:
Can you also ensure there's a test for an instantiation like `func<-1>();`? If the test is expected to cause errors, it should be a SemaSYCL test, but if the code is accepted, there should be both a SemaSYCL test (for the warning) and a CodeGenSYCL test (for the codegen).

Reply:
Thanks @AaronBallman for looking into the patch.

>>Can you also ensure there's a test for an instantiation like func<-1>();
Test is added

All attributes are expected to cause errors. So SemaSYCL tests are added here.
[intel::max_global_work_dim)]] 
[[intel::num_simd_work_items()]] 
[[intel::reqd_sub_group_size()]]